extensions/local_rate_limit/plugin.cc

```diff
@@ -0,0 +1,139 @@
+#include "extensions/local_rate_limit/plugin.h"
+
+#include "absl/strings/str_cat.h"
+#include "extensions/common/wasm/json_util.h"
+#include "extensions/local_rate_limit/bucket.h"
+
+using ::nlohmann::json;
+using ::Wasm::Common::JsonArrayIterate;
+using ::Wasm::Common::JsonGetField;
+using ::Wasm::Common::JsonObjectIterate;
+using ::Wasm::Common::JsonValueAs;
+
+namespace {
+
+// tooManyRequest returns a 429 response code.
+void tooManyRequest() { sendLocalResponse(429, "Too many requests", "", {}); }
+
+bool shouldRateLimit() {
+  /*
+    Add any rate limit condition here, such as host, path, etc.
+  */
+  return true;
+}
+
+}  // namespace
+
+static RegisterContextFactory register_LocalRateLimit(
+    CONTEXT_FACTORY(PluginContext), ROOT_FACTORY(PluginRootContext));
+
+bool PluginRootContext::onConfigure(size_t configuration_size) {
+  if (!parseConfiguration(configuration_size)) {
+    return false;
```

Comment:
Just to be sure, a return false here does not crash wasm, right ?

Reply:
Right, the host now has a `canary` step which brings up a one time VM, run it with the provided configuration, and rejects xds  update if `onConfigure` fails there.